packages/datetime/test/dateTimePickerTests.tsx

```diff
@@ -101,6 +100,14 @@ describe("<DateTimePicker>", () => {
                 assert.isFalse(getSelectedDay().exists());
             });
         }
+
+        it("Passing the same value prop twice does not clear the selected date", () => {
+            const { root, getSelectedDay } = wrap(<DateTimePicker />);
+            assert.isTrue(getSelectedDay().exists());
+            root.setProps({ value: undefined });
```

Comment:
But that does not test the same thing. I think it is more appropriate to change the name of this test to reflect what it's really testing (which I did). So the DateTimePicker only selects its own date and time if the `value` prop is undefined. What we are trying to fix here is given that a user implements `onChange` but does NOT pass in a `value`, we want to avoid the `onChange` handler, which will pass in new props, to interfere with / clear out the date/time state managed by the `DateTimePicker`. 

For example, this test should fail: 

```        it("Passing the same value prop twice does not change a user-selected date", () => {
            const value = new Date(2012, 2, 5, 6, 5, 40);
            const { root, getDay } = wrap(<DateTimePicker value={value} />);
            assert.strictEqual(root.state("dateValue"), value);
            
            getDay(15).simulate("click");

            const modifiedDate = new Date(2012, 2, 15, 6, 5, 40);
            assert.strictEqual(root.state("dateValue"), modifiedDate); // this will fail because DateTimePicker no longer manages state since value is defined
            
            root.setProps({ value: value });
            assert.strictEqual(root.state("dateValue"), modifiedDate);
        });
```